### DISCUSSION TITLE
[Feature] Utility which shows biggest files

### DISCUSSION THREAD

**Discussion by KuramaSyu** [immich-app/immich#12717] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

A feature which is under utilities next to duplicates, which checks the biggest files. What it most likely should find, are big video files where the owner can then decide "do I really need this 4GB video?". Maybe a method for compressing big files to replace them could go there too.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by shad0wrunner** [comment]
Upvoting. Such function present in G**gle Ph*tos as well. It is very useful to free disk space by eliminating old and big videos. Additional option could be to suggest and transcode video to a smaller size format, selected by user.

**Comment by wireva** [comment]
I would love this feature as well. I may start working on a PR.
For now you can get the largest assets with this SQL query:
```sql
select * FROM exif ORDER BY "fileSizeInByte" DESC LIMIT 10;
```

then simply open any asset in the browser and paste the corresponding UUID in the URL.

**Reply by wireva** [reply]
https://github.com/immich-app/immich/pull/18040
